pykrige/kriging_tools.py

```diff
@@ -246,3 +248,181 @@ def read_asc_grid(filename, footer=0):
         cellsize = (dx, dy)
 
     return grid_array, x, y, cellsize, no_data
+
+
+def write_zmap_grid(x, y, z, filename="output.zmap", no_data=-999.0, coord_sys='<null>'):
+    r"""Writes gridded data to ASCII grid file in zmap format (\*.zmap).
+
+    This is useful for exporting data to a GIS program, or Petrel
+    https://gdal.org/drivers/raster/zmap.html
+
+    Parameters
+    ----------
+    x : array_like, shape (N,) or (N, 1)
+        X-coordinates of grid points at center of cells.
+    y : array_like, shape (M,) or (M, 1)
+        Y-coordinates of grid points at center of cells.
+    z : array_like, shape (M, N)
+        Gridded data values. May be a masked array.
+    filename : string, optional
+        Name of output \*.zmap file. Default name is 'output.zmap'.
+    no_data : float, optional
+        no data value to be used
+    coord_sys : String, optional
+        coordinate sytem description
+    """
+
+    nodes_per_line = 5
+    field_width = 15
+
+    if np.ma.is_masked(z):
+        z = np.array(z.tolist(no_data))
+
+    x = np.squeeze(np.array(x))
+    y = np.squeeze(np.array(y))
+    z = np.squeeze(np.array(z))
+    nx = len(x)
+    ny = len(y)
+
+    dx = abs(x[1] - x[0])
+    dy = abs(y[1] - y[0])

```

Comment:
Could you add the same check like in the asc routine for checking the grid?
```python
    if (
                    not np.isclose(abs((x[-1] - x[0]) / (x.shape[0] - 1)), dx)
                    or not np.isclose(abs((y[-1] - y[0]) / (y.shape[0] - 1)), dy)
            ):
                raise ValueError(
                    "X or Y spacing is not constant; *.zmap grid cannot be written."
                )
```
to address #197?